packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts

```diff
@@ -222,6 +197,67 @@ export class WebRtcCommunicationChannel implements CommunicationChannel {
     this.statisticsSubject.next(cloneDeep(this.statistics));
   }
 
+  /**
+   * If there are already known TURN servers, this is a noop.
+   * Otherwise wait for TURN servers from the Widget API until a timeout.
+   */
+  private async initTurnServers(): Promise<void> {
+    if (this.turnServers !== undefined) {
+      // TURN servers already known
+      return;
+    }
+
+    const source = race(
+      interval(250).pipe(
+        map(() => {
+          return this.turnServers;
+        }),
+        filter((t) => t !== undefined),
+      ),
+      NEVER.pipe(timeout(2500)),
+    );
+
+    await firstValueFrom(source);
+  }
+
+  private async handleSessionJoined(session: Session): Promise<void> {
+    const sessionId = this.sessionManager.getSessionId();
+
+    this.logger.log('joined', session.sessionId, session.userId);
+
+    if (!sessionId) {
+      throw new Error('Unknown session id');
+    }
+
+    // Wait for the Widget API or TURN servers to be ready
```

Comment:
wait for the Widget API AND turn servers to be ready?

Reply:
Done